cmd/gokr-updater/updater.go

```diff
@@ -41,7 +54,28 @@ func main() {
 		log.Fatal(err)
 	}
 	baseUrl.Path = "/"
+	httpClient, foundMatchingCertificate, err := httpclient.GetTLSHttpClientByTLSFlag(useTLS, baseUrl)
+	probeResp, err := httpClient.Get("http://" + baseUrl.Host)
+	if err != nil {
+		log.Fatalf("probing url for https: %v", err)
+	}
+	if probeResp.TLS != nil {
+		baseUrl.Scheme = "https"
+		*update = baseUrl.String()
+	}
 
+	if baseUrl.Scheme != "https" && foundMatchingCertificate {
+		fmt.Printf("\n")
+		fmt.Printf("!!!WARNING!!! Possible SSL-Stripping detected!\n")
+		fmt.Printf("Found certificate for hostname in your client configuration but the host does not offer https!\n")
+		fmt.Printf("\n")
+		fmt.Printf("Continue anyway? (type 'yes' continue)\n")
+		reader := bufio.NewReader(os.Stdin)
```

Comment:
Let’s use a flag, e.g. `-insecure`, for consistency here (we don’t have interactive prompts anywhere else).